src/Utils/util.ts

```diff
@@ -93,4 +94,15 @@ export function earlierDateOrTimeToday(timestamp: string): string {
     const dialogTime = moment(timestamp)
     const isDialogCreatedToday = dialogTime.diff(endOfYesterday) >= 0
     return dialogTime.format(isDialogCreatedToday ? 'LTS' : 'L')
+}
+
+export function isNewActionUnique(newAction: CLM.ActionBase, actions: CLM.ActionBase[]): boolean {
+    const needle = normalizeActionAndStringify(newAction)
+    const haystack = actions.map(action => normalizeActionAndStringify(action))
+    return haystack.findIndex(straw => straw === needle) < 0
+}
+
+function normalizeActionAndStringify(newAction: CLM.ActionBase) {
```

Comment:
Hmm, not a fan of the word normalize but I could see it.  Will see what others say.

Options, but I would prefer something like `removeIncomparableProperties`

Also would keep the functions split. You generally wnt things to be composable and to small things well.  I may want a stripped action but not the string version of it.  E.g. `removeIncomparableProperties(stringify(action))`

Reply:
I'm inclined to leave this one as coded. The composable and small philosophy is a worthy point going forward, especially when writing new code to the code base.

Reply:
Ok.  I didn't notice before, but rename `newAction` parameter to just `action`. This function is used on the set of existing actions as well.